src/v3/plugins/github/address.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+
+import {NodeAddress, type NodeAddressT} from "../../core/graph";
+
+export opaque type GithubAddressT: NodeAddressT = NodeAddressT;
+
+const GITHUB_PREFIX = NodeAddress.fromParts(["sourcecred", "github"]);
+function githubAddress(...parts: string[]): GithubAddressT {
+  return NodeAddress.append(GITHUB_PREFIX, ...parts);
+}
+
+export type RepoAddress = {|
+  +type: "REPO",
+  +owner: string,
+  +name: string,
+|};
+export type IssueAddress = {|
+  +type: "ISSUE",
+  +repo: RepoAddress,
+  +number: string,
+|};
+export type PullAddress = {|
+  +type: "PULL",
+  +repo: RepoAddress,
+  +number: string,
+|};
+export type ReviewAddress = {|
+  +type: "REVIEW",
+  +pull: PullAddress,
+  +fragment: string,
+|};
+export type CommentAddress = {|
+  +type: "COMMENT",
+  +parent: IssueAddress | PullAddress | ReviewAddress,
+  +fragment: string,
+|};
+export type UserlikeAddress = {|
+  +type: "USERLIKE",
+  +login: string,
+|};
+
+export type StructuredAddress =
+  | RepoAddress
+  | IssueAddress
+  | PullAddress
+  | ReviewAddress
+  | CommentAddress
+  | UserlikeAddress;
+
+export function structure(x: GithubAddressT): StructuredAddress {
+  if (!NodeAddress.hasPrefix(x, GITHUB_PREFIX)) {
+    throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+  }
+  const [_unused_sc, _unused_gh, kind, ...rest] = NodeAddress.toParts(x);
```

Comment:
There is no explicit length check here. The code still behaves correctly
(if the length is too short, then `kind` will be `undefined`, which hits
the `default` case of the `switch`), but if you want to add the explicit
check, I’d support that. I’m also fine with not including it.

Reply:
Going to leave it as-is.